### PR TITLE
Maya/187620 invite email input

### PR DESCRIPTION
**Pull request by mmmmmmaya**
### Summary
- **What:** Add UI for inviting new users
- **Ticket:** [[sc-187620]](https://app.shortcut.com/genepi/story/187620)
- **Env:** https://maya-invitedialog-frontend.dev.czgenepi.org/data/samples
- **Design:** https://www.figma.com/file/SALQZGfIJntBes0HvIzVeX/Onboarding-New-Users-V0?node-id=280%3A48892

### Demos
after

### Known outstanding tasks
- Button doesn't currently send invites
- The header should have the group name bolded, but sds only allows simple string in DialogTitle, so a PR will be required before that can be implemented
- No debounce on the input validation yet
- In the example rdev, the modal is permanently open. The actual modal will not be like this.

### Checklist
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I either asked design for a review or hid my changes behind a feature flag
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

**Review by Janeece — APPROVED** [review]
@mmmmmmaya this looks great, no comments!

File: src/frontend/src/common/constants/inputDelimiters.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const INPUT_DELIMITERS = /[\n\t,]/g;
```
Comment (vincent-czi) [inline src/frontend/src/common/constants/inputDelimiters.ts:1]: This isn't exactly about the delimiters, but I noticed some stuff regarding spaces. On the current version in the rdev, if I have just a single line of this:
```
  iHave2SpacesOnEitherSide@hi.com  
```
then hit the Send Invites button, it doesn't complain at all.

But if I try to use this
```
  iHave2SpacesOnEitherSide@hi.com  
    iHave4SpacesOnEitherSide@hi.com    
noSpacesOnMyLeftBut2OnMyRight@hi.com  
```
and hit the send button, I get 2 errors, but the final entry ("noSpaces...") is valid.


Maybe we could just remove all space characters in the entry box entirely before parsing it for email addresses and validity?

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/components/SentNotification/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { pluralize } from "src/common/utils/strUtils";
+import Notification from "src/components/Notification";
+
+interface Props {
+  onDismiss(): void;
+  open: boolean;
+  numSent: number;
+}
+
+const SentNotification = ({ numSent, onDismiss, open }: Props): JSX.Element => {
```
Comment (vincent-czi) [inline src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/components/SentNotification/index.tsx:13]: it seems odd that this notification lives under `InvalidEmailError` in the dir structure. Nothing about it is invalid, right?

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/style.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ${(props) => {
+    const spaces = getSpaces(props);
+
+    // TODO (mlila): overlay colors not yet available from sds,
+    // TODO          so hardcoding this for now.
+    return `
+      padding: ${spaces?.l}px;
+
+      :nth-of-type(odd) {
+        background-color: #F6EAEA;
```
Comment (vincent-czi) [inline src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/style.ts:14]: Not saying you got to do the same thing, but what I did elsewhere for creating zebra stripes in a similar table kind of situation was
```
background-color: rgba(0, 0, 0, 0.04);
```
which effectively dimmed the background color a smidge so I didn't have to worry about tracking two color vars for the stripe and the surrounding div

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      !hasMoreThan50 && newInvalidAddresses.length === 0 && addresses.length > 0
+    );
+  };
+
+  const onInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setInputValue(value);
+  };
+
+  // const title = (
```
Comment (vincent-czi) [inline src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx:57]: Should this comment be removed?

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  //   <StyledSpan>
+  //     Invite to <B>Santa Clara County</B>
+  //   </StyledSpan>
+  // );
+
+  const instructions = [
+    <span key={0}>You can send a maximum of 50 invitations at a time.</span>,
+    <span key={1}>Separate emails by tabs, commas, or enter one per row.</span>,
+    <span key={2}>
+      Members have full access to group samples and phyogenetic trees; they are
```
Comment (vincent-czi) [inline src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx]: phyogenetic --> phy**l**ogenetic

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            label="Emails"
+            onChange={onInputChange}
+            placeholder="e.g. userone@domain.com, usertwo@domain.com"
+            intent={inputIntent}
+            value={inputValue}
+            maxRows={4}
+          />
+          {hasMoreThan50Invites && (
+            <StyledCallout intent="error">
+              <B>You can’t send more than 50 invites at a time. </B>
```
Comment (vincent-czi) [inline src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx:110]: Do we use directional apostrophes/quotes? For some reason I thought we just used straight ones. I don't have a strong opinion here, just want to verify this was intentional

**Review by vincent-czi — APPROVED** [review]
Overall looks good! I've got a couple comments that would be good to address, but I also know this isn't immediately going to go live -- if you just threw them on to a todo list and planned to get around to them eventually in a later iteration, I've got no problem with that.

Nice work! Very easy to read!

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/components/SentNotification/index.tsx (hunk 2 above)
Comment (mmmmmmaya) [reply src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/components/InvalidEmailError/components/SentNotification/index.tsx:13]: oops, that was a mistake. I'll fix in follow up

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx (hunk 4 above)
Comment (mmmmmmaya) [reply src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx:57]: nope, I'm going to use it in my next PR

File: src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx (hunk 6 above)
Comment (mmmmmmaya) [reply src/frontend/src/views/GroupMembersPage/components/MembersTab/components/InviteModal/index.tsx:110]: I literally copy paste the text from Figma, so whatever Janeece uses is what goes into the app

**Pull request merged**